SPEC.md

```diff
@@ -155,6 +155,37 @@ When persisting an output artifact to a file system, if the build tool has addit
 it should persist that metadata to a subdirectory of the directory to which the output artifact is being written of the form: ```.bom/metadata/${tool}/```.  Filenaming and subdirectory structure below that point is at the discretion of the build tool.
 ## Annex A
 
+## Annex B
+
+### Build tool selection of gitoid type
+
+GITOID_BLOB_TYPE should be set in order of precedence by:
+1. A non-empty environment variable named GITOID_BLOB_TYPE.
+2. A build tool specific flag.
+The value of GITOID_BLOB_TYPE should contain the gitoid type. It may contain multiple gitoid types separated by comma (e.g., sha1,sha256). The build tool should generate the gitoids for all specified types. The build tool should implement a default gitoid type that should be used when no flag is provided.
+The build tool should report an error if the requested gitoid type is not supported.
+
+
+### Gitoid persistence by a Build Tool in ELF Objects/Executables
+When persisting the gitoid to an ELF object or an ELF executable, the build tool should create a section ```.note.gitbom``` of type SHT_NOTE and place the gitoid in the descriptor field of the Note entry. Multiple Note entries should be created, one for each gitoid type when multiple gitoid types are involved. Each Note entry must contain the following fields in the same order as given below:
+1. namesz (4 bytes): This field must be set to a value of 8, the length of the 'owner' field (including padding) in bytes.
```

Comment:
although it is padded to 8 bytes, I think the value of namesz should be 7, not 8. "GITBOM\0" length is 7.